### PR TITLE
Get sliprefix relative to the dynamic library

### PR DESCRIPTION
**Pull request by isuruf**
Fixes https://github.com/nest/nest-simulator/issues/2237

cc @steffengraber

### CODE REVIEW

**Comment by steffengraber** [comment]
@jougs and @terhorstd Could you take a look at this, please? This could solve the problems with the conda packages.

**Comment by jougs** [comment]
@isuruf: can you please look into the [failing static tests](https://github.com/nest/nest-simulator/runs/5507893387?check_suite_focus=true#step:9:313)? It is usually sufficient to install `clang-format` version 9 and run `build_support/format_all_c_c++_files.sh` in the repository to get the C++ formatting right.

File: sli/slistartup.cc [hunk 1]
```diff
@@ -51,6 +51,13 @@
 extern char** environ;
 #endif
 
+#if defined(__linux__) || defined(__APPLE)
```
Comment (heplesser) [inline sli/slistartup.cc:54]:
```suggestion
#if defined(__linux__) || defined(__APPLE__)
```

File: sli/slistartup.cc [hunk 2]
```diff
@@ -133,9 +140,25 @@ SLIStartup::GetenvFunction::execute( SLIInterpreter* i ) const
   i->EStack.pop();
 }
 
+static std::string getsliprefix() {
+#if defined(__linux__) || defined(__APPLE)
+    Dl_info info;
+    if (dladdr((void*)getsliprefix, &info)) {
+        char const * soname = info.dli_fname;
+        std::string result = std::string(soname);
```
Comment (heplesser) [inline sli/slistartup.cc:148]:
```suggestion
       std::string result( info.dli_fname );
```
Can you find a more informative name than `result`?

File: sli/slistartup.cc [hunk 3]
```diff
@@ -133,9 +140,25 @@ SLIStartup::GetenvFunction::execute( SLIInterpreter* i ) const
   i->EStack.pop();
 }
 
+static std::string getsliprefix() {
```
Comment (heplesser) [inline sli/slistartup.cc:143]: Please add a doxygen comment explaining what this function does.

**Review by heplesser — CHANGES_REQUESTED** [review]
Please see my detailed comments.

**Comment by jougs** [comment]
@heplesser: can you please come to the hackathon? we're jointly working on this right now!

**Comment by jougs** [comment]
Closing in favor of more generic fix in #2348. Anyway, thanks for your contribution!

**Pull request closed without merging**